language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/imports/AddModuleToBallerinaTomlCodeAction.java

```diff
@@ -0,0 +1,179 @@
+
+    public static final String NAME = "Add Module to Ballerina.toml";
+    
+    @Override
+    public boolean validate(Diagnostic diagnostic, DiagBasedPositionDetails positionDetails,
+                            CodeActionContext context) {
+        return CodeActionNodeValidator.validate(context.nodeAtRange());
+    }
+
+    @Override
+    public List<CodeAction> getCodeActions(Diagnostic diagnostic,
+                                           DiagBasedPositionDetails positionDetails,
+                                           CodeActionContext context) {
+        if (!diagnostic.diagnosticInfo().code().equals(DiagnosticErrorCode.MODULE_NOT_FOUND.diagnosticId())) {
+            return Collections.emptyList();
+        }
+        Optional<Project> project = context.workspace().project(context.filePath());
+        if (project.isEmpty()) {
+            return Collections.emptyList();
+        }
+        Optional<BallerinaToml> toml = project.get().currentPackage().ballerinaToml();
+        if (toml.isEmpty()) {
+            return Collections.emptyList();
+        }
+        
+        Optional<String> msg = positionDetails.diagnosticProperty(0);
+        if (msg.isEmpty()) {
+            return Collections.emptyList();
+        }
+        
+        // Consider orgname empty case
+        String[] orgAndRest = msg.get().split("/");
+        if (orgAndRest.length != 2) {
+            return Collections.emptyList();
+        }
+        String org = orgAndRest[0];
+        String[] moduleAndPrefix = orgAndRest[1].split(" as ");
+        List<PackageVersion> loadedPackageVersions = new ArrayList<>();
+        String pkg = getPkgAndVersions(LSPackageLoader.getInstance(context.languageServercontext()), project.get(), 
+                org, moduleAndPrefix[0], loadedPackageVersions);
+        if (loadedPackageVersions.isEmpty()) {
+            return Collections.emptyList();
+        }
+        
+        Position dependencyStart = new Position(getDependencyStartLine(toml.get()), 0);
+        String dependency = String.format("[[dependency]]%norg = \"%s\"%nname = \"%s\"%nversion = " +
+                "\"%s\"%nrepository = \"local\"%n%n", org, pkg, getLatestVersion(loadedPackageVersions));
+        TextEdit textEdit = new TextEdit(new Range(dependencyStart, dependencyStart), dependency);
+        CodeAction action = CodeActionUtil.createCodeAction(CommandConstants.ADD_MODULE_TO_BALLERINA_TOML, 
+                List.of(textEdit), project.get().sourceRoot().resolve(ProjectConstants.BALLERINA_TOML).toString(), 
+                CodeActionKind.QuickFix);
+        return Collections.singletonList(action);
+    }
+    
+    private String getPkgAndVersions(LSPackageLoader lsPackageLoader, Project project, String org, 
+                                     String moduleWithPkg, List<PackageVersion> versions) {
+        String names = moduleWithPkg;
+        while (true) {
+            int i = names.lastIndexOf(".");
```

Comment:
We can argue the same thing in the other way as well. 

Another thing is, let's say we have two packages as shown below.
first package-
x
|_y

second package-
x.y
|_mod1

We have an import statement like `import ballerina/x.y.mod1`. In this case, we have to add the second package to Ballerina.toml. If we extract the package from lhs, we pick the first one which is correct. 

Because of these reasons I think the existing way is better.

Reply:
+1